polyfactory/factories/base.py

```diff
@@ -665,6 +666,31 @@ def get_model_fields(cls) -> list[FieldMeta]:  # pragma: no cover
         """
         raise NotImplementedError
 
+    @classmethod
+    def get_factory_fields(cls) -> list[tuple[str, Any]]:
+        factory_fields = cls.__dict__.items()
+        return [
+            (field_name, field_value)
+            for field_name, field_value in factory_fields
+            if not (field_name.startswith("__") or field_name == "_abc_impl")
+        ]
+
+    @classmethod
+    def _check_declared_fields_exist_in_model(cls) -> None:
+        model_fields_names = {field_meta.name for field_meta in cls.get_model_fields()}
+        factory_fields = cls.get_factory_fields()
+
+        for field_name, field_value in factory_fields:
+            if field_name in model_fields_names:
+                continue
+
+            error_message = (
+                f"{field_name} is declared on the factory {cls.__name__}"
+                f" but it is not part of the model {cls.__model__.__name__}"
+            )
+            if isinstance(field_value, (Use, PostGenerated, Ignore, Require)):
+                raise ConfigurationException(error_message)
```

Comment:
Looking at this again, I'm actually a bit confused on whether this should raise an exception. Technically, this would be a breaking change.

Maybe we could add a `__check_model__` field or something which, if set to `True`, it would raise an error with the default being `False`. If the value is `False` then we give a warning instead. Then in v3, we could remove all of this and just throw an error. 

What do you think?

Reply:
Those will only reveal bugs (AKA no-op field definitions), right?

Do you identify cases where this would have been done on purpose?

On the principle I agree that this is a breaking change, I can implement the `__check_model__` flag even though it feels a bit like overkill to me :slightly_smiling_face:

Reply:
You're absolutely right that in 99.999% of the cases, this would be a bug in the user's code. However, I still don't feel comfortable breaking compatibility when there's a workaround (a more annoying one and might even be overkill like you said).

Reply:
Ok I've done the change :+1: :slightly_smiling_face: